ts/Core/Axis/Color/ColorAxis.ts

```diff
@@ -837,6 +855,18 @@ class ColorAxis extends Axis implements AxisLike {
 
 }
 
+addEvent(ColorAxis, 'afterSetScale', function (): void {
+    if (this.chart.legend.options.enabled) {
+        const symbolBox = this.legendItem?.symbol?.getBBox(),
+            actualLen = this.horiz ? symbolBox?.width : symbolBox?.height;
+        // Fire update if dimensions have changed, but in another place, than by
+        // updating color axis
+        if (actualLen && actualLen !== this.len) {
+            this.update({});
```

Comment:
I figured out, that my idea about the event could be bad and we do not need any logic in special events. Instead of this, we can simply overwrite attributes of the existing color axis symbol.